code/game/objects/structures/table_frames.dm

```diff
@@ -31,15 +31,15 @@
 		return
 
 	var/obj/item/stack/material = I
-	if (istype(I, /obj/item/stack))
+	if (istype(material))
 		if(material?.tableVariant)
 			if(material.get_amount() < 1)
 				to_chat(user, "<span class='warning'>You need one [material.name] sheet to do this!</span>")
 				return
 			to_chat(user, "<span class='notice'>You start adding [material] to [src]...</span>")
 			if(do_after(user, 20, target = src) && material.use(1))
 				make_new_table(material.tableVariant)
-		else
+		else if(istype(material, /obj/item/stack/sheet))
```

Comment:
Shouldn't this be just metal ?

Reply:
No,  you can make tables out of any material

Reply:
so why doesn't stack just have set variant instead of this ?

Reply:
Well?

Reply:
@AnturK @Qustinnus this still needs resolving

Reply:
my bad I didn't see this pop up in my mails. I'm not sure if I' m understanding you correctly, but tables are greyscale and just color themselves, why would they have a set variant? It's supposed to be dynamic.

Reply:
As in why this needs a special handling. It seems having a default variant set on base sheet type would do exactly the same thing without the need for any special cases. Or did you mean type == /obj/item/stack/sheet here.

Reply:
This needs special handling because if it's not a sheet we don't want it doing anything, to prevent rods and tiles being used as a building material (They are made of materials)